src/fhirtypes/ElementDefinition.ts

```diff
@@ -1889,6 +1892,73 @@ export class ElementDefinition {
     return [];
   }
 
+  /**
+   * Returns an array of an ElementDefinition's unfolded children.
+   * @param {ElementDefinition} targetElement - The ElementDefinition being unfolded
+   * @returns {ElementDefinition[]} An array of the targetElement's children, with the IDs altered and
+   * the original property re-captured.
+   */
+
+  private cloneChildren(targetElement: ElementDefinition): ElementDefinition[] {
+    return targetElement?.children().map(e => {
+      const eClone = e.clone();
+      eClone.id = eClone.id.replace(targetElement.id, this.id);
+      eClone.structDef = this.structDef;
+      eClone.captureOriginal();
+      return eClone;
+    });
+  }
+
+  /**
+   * Checks a StructureDefinition's differential to determine if the profile element extension has been used on
+   * an element.
+   * @param {Object} profileJson - The json representation of this ElementDefinition's structDef
+   * @returns {boolean} True if the profile element extension is found on this elements profile property, false
+   * if the extension is not found
+   */
+  private hasProfileElementExtension(profileJson: any): boolean {
+    const elementName = this.getContentReferenceId();
+
+    const elementType = profileJson.differential.element.find(
+      (element: any) => element.id === elementName
+    )?.type?.[0];
+
+    // If the type property is not present in the differential, the profile-element extension is not present
+    if (!elementType) {
+      return false;
+    }
+
+    let profileCanonical, extensionUrl, targetElement: string;
+
+    // If the profile and _profile arrays are not present, the profile-element extension is not present
+    if (elementType.profile && elementType._profile) {
+      const profileIndex = elementType._profile.findIndex(
+        (profile: any) =>
+          profile &&
+          profile.extension.find(
+            (extension: any) =>
+              extension.hasOwnProperty('url') && extension.hasOwnProperty('valueString')
+          ) !== undefined
```

Comment:
This can be shortened a bit, since we don't need to actually use the found value:
```
(profile: any) => profile.extension?.some( ... )
```
Alternatively, since we do eventually want the index of that extension, it may be possible to rewrite this and the following search in order to grab both indices at the same time.

Reply:
Thank you! I've switched over to using the `some` function in commit `5504e91`, though I did not combine the two searches into one since I don't think either `some` or `findIndex` actually allow for values to be set in their callback functions.